### PR TITLE
Handle string error codes

### PR DESCRIPTION
**Pull request by fogelito**
PDOExeptions error code are string , for example:
SQLSTATE[HY000]: General error: 2006 MySQL server has gone away.

error code is = HY000

## What does this PR do?

(Provide a description of what this PR does and why it's needed.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Screenshots may also be helpful.)

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [ ] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

**Review by eldadfux — APPROVED** [review]

File: src/Appwrite/Extend/Exception.php [hunk 1]
```diff
@@ -296,11 +296,17 @@ class Exception extends \Exception
...
 
-    public function __construct(string $type = Exception::GENERAL_UNKNOWN, string $message = null, int $code = null, \Throwable $previous = null)
+    public function __construct(string $type = Exception::GENERAL_UNKNOWN, string $message = null, $code = null, \Throwable $previous = null)
     {
         $this->errors = Config::getParam('errors');
         $this->type = $type;
         $this->code = $code ?? $this->errors[$type]['code'];
+
+        // todo: Handle better PDOExceptions or string errors
+        if(is_string($this->code)) {
```
Comment (eldadfux) [inline src/Appwrite/Extend/Exception.php:306]: Not sure we want to overwrite this.

Comment (fogelito) [reply src/Appwrite/Extend/Exception.php:306]: Got it , so perhaps we need to find a different solutions with the PDOExceptions string error code  🤔

Comment (abnegate) [inline src/Appwrite/Extend/Exception.php:306]:
```suggestion
        if(\is_string($this->code) && !\is_numeric($this->code)) {
```

File: src/Appwrite/Extend/Exception.php [hunk 2]
```diff
@@ -296,11 +296,17 @@ class Exception extends \Exception
...
     protected bool $publish;
 
-    public function __construct(string $type = Exception::GENERAL_UNKNOWN, string $message = null, int $code = null, \Throwable $previous = null)
+    public function __construct(string $type = Exception::GENERAL_UNKNOWN, string $message = null, $code = null, \Throwable $previous = null)
     {
         $this->errors = Config::getParam('errors');
         $this->type = $type;
         $this->code = $code ?? $this->errors[$type]['code'];
+
+        // todo: Handle better PDOExceptions or string errors
```
Comment (abnegate) [inline src/Appwrite/Extend/Exception.php:305]:
```suggestion
```

File: src/Appwrite/Extend/Exception.php [hunk 3]
```diff
@@ -296,11 +296,16 @@ class Exception extends \Exception
...
-    public function __construct(string $type = Exception::GENERAL_UNKNOWN, string $message = null, int $code = null, \Throwable $previous = null)
+    public function __construct(string $type = Exception::GENERAL_UNKNOWN, string $message = null, $code = null, \Throwable $previous = null)
     {
         $this->errors = Config::getParam('errors');
         $this->type = $type;
         $this->code = $code ?? $this->errors[$type]['code'];
+
+        if(\is_string($this->code) && !\is_numeric($this->code)) {
+            $this->code = 500;
+        }
```
Comment (abnegate) [inline src/Appwrite/Extend/Exception.php:307]:
```suggestion
        if(\is_string($this->code)) {
            if (\is_numeric($this->code)) {
                $this->code = (int) $this->code;
            } else {
                $this->code = 500;
            }
        }
```

File: src/Appwrite/Extend/Exception.php [hunk 4]
```diff
@@ -296,11 +296,20 @@ class Exception extends \Exception
     protected array $errors = [];
     protected bool $publish;
 
-    public function __construct(string $type = Exception::GENERAL_UNKNOWN, string $message = null, int $code = null, \Throwable $previous = null)
+    public function __construct(string $type = Exception::GENERAL_UNKNOWN, string $message = null, $code = null, \Throwable $previous = null)
```
Comment (abnegate) [inline src/Appwrite/Extend/Exception.php:299]:
```suggestion
   public function __construct(string $type = Exception::GENERAL_UNKNOWN, string $message = null, int|string $code = null, \Throwable $previous = null)
```

**Review by abnegate — APPROVED** [review]

**Pull request merged**